packages/react-magma-dom/src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { ThemeContext } from '../../theme/ThemeContext';
+import { Modal, ModalProps } from '../Modal';
+import { TransitionProps } from '../Transition';
+
+export enum DrawerPosition {
+  top = 'top',
+  bottom = 'bottom',
+  left = 'left',
+  right = 'right',
+}
+
+const transitionPreset: {
+  [key in DrawerPosition]: Omit<TransitionProps, 'isOpen'>;
+} = {
+  top: { slideTop: true },
+  bottom: { slideBottom: true },
+  left: { slideLeft: true },
+  right: { slideRight: true },
+};
+
+/**
+ * @children required
+ */
+export interface DrawerProps extends ModalProps {
+  testId?: string;
+  /**
+   * Style properties for the drawer
+   */
+  style?: React.CSSProperties;
```

Comment:
I wonder if we should call this property `drawerStyle` to distinguish it from `containerStyle`.  I believe that's what we do in other components that take multiple style props.